app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt

```diff
@@ -85,6 +82,10 @@ import javax.inject.Singleton
 )
 class AppVersionActivityTest {
 
+  @get:Rule
+  val activityScenarioRule: ActivityScenarioRule<AppVersionActivity> =
+    ActivityScenarioRule(createAppVersionActivityIntent())
```

Comment:
launch function internally using ActivityScenario and calling ` scenario.launchInternal();`. 

Currently, we use ActivityScenario and this implementation is of ActivityScenarioRule, need to check the difference as I guess there is a tiny difference between them that,
ActivityScenario doesn't clean up the state after test ends but ActivityScenarioRule does it. 

Also, in the `ActivityScenarioRule`, we cannot do any kind of setup before launching the activity, that's something we can do with `ActivityScenario`. 

So if wee move with `ActivityScenario` and don't use `.use{}`, then we need something like

```kotlin
@After
fun cleanup() {
    scenario.close()
}
```

Need some expert points here @BenHenning

Reply:
We talked about this during the team meeting today @anandwana001. I suggest looking over the team notes; there were some bits in there that were recorded from the discussion I think.

@rt4914 I'm curious about one thing: does scenario rule let us handle configuration changes correctly? I was under the impression that the scenario gets recreated when the activity does (since scenarios treat DESTROYED as a terminal state). Can you confirm?

Reply:
I have made two new commits.
Commit https://github.com/oppia/oppia-android/pull/2783/commits/9d77057fed5b97e5cd99cb7c8d223d76c8305cdd shows that config change can easily be done with `ActivityScenarioRule`.

Commit https://github.com/oppia/oppia-android/pull/2783/commits/d9fd2989c7e4e420d303db1aa31bd35d1a7c8446 shows that the code which we need before starting the activity also works correctly with `ActivityScenarioRule`.

Overall there is an **advantage** of using `ActivityScenarioRule`: 
(a.) It can replace both `ActivityTestRule` and `ActivityScenario` and can provide consistency to all tests.
(b.) ActivityScenario does't clean up device state automatically and may leave the activity keep running after the test finishes. Call close() in your test to clean up the state or use try-with-resources statement. This is optional but highly recommended to improve the stability of your tests. Also, consider using ActivityScenarioRule. [Reference](https://developer.android.com/reference/androidx/test/core/app/ActivityScenario)

**Limitation** with `ActivityScenarioRule`:
**It does not allow us to change intent inside tests which we normally does for various test cases.**

**Solution** is to use `LazeActivityScenarioRule` which can allow us to change the intent for each test case. [Reference](https://medium.com/stepstone-tech/better-tests-with-androidxs-activityscenario-in-kotlin-part-1-6a6376b713ea)

**Open Questions**
If we use `LazeActivityScenarioRule` can this create some other type of issue?
WDYT we should do finally?

Also, I highly recommend reading [this blog](https://medium.com/stepstone-tech/better-tests-with-androidxs-activityscenario-in-kotlin-part-1-6a6376b713ea) as it summaries the difference between `ActivityTestRule`, `ActivityScenario` and `ActivityScenarioRule` nicely.

Reply:
Please keep in mind we can't just copy LazyActivityScenarioRule--we would need to create our own rule that effectively does the same thing as ActivityScenarioRule but lets us customize the intent. I think the need here is pretty clear, but we need to define specifically what the desired API is. I suspect defaulting the intent is actually desirable for most situations, and having per-test overriding seems reasonable. However, it's not straightforward to implement that without accidentally opening multiple activities. I think we would need to start a draft PR introducing such a utility & iterate on the design a bit there to figure out something that would work for all cases. I do not recommend referencing the one from the blog--its existence introduces the idea that we can try to create on our own, but we need to be more careful about overly depending on external code that's not pre-packaged into a library.

Also, I'm not sure my original question regarding configuration changes was answered. See #2572 for specifics here--the perform orientation change doesn't actually force configuration changes in all environments (e.g. Robolectric). To answer this question, I think we need to check a test that passes in #2572 for both Espresso & Robolectric with the new utility, but also using ActivityScenarioRule.

Reply:
@BenHenning Considering that this requires more research work now, can I pass in to @anandwana001 and focus back on A11y PRs. WDYT?

Reply:
Yup, sgtm @rt4914.

Reply:
Thanks.